boot/assets.go

```diff
@@ -50,19 +151,47 @@ type TrustedAssetsInstallObserver struct {
 //
 // Implements gadget.ContentObserver.
 func (o *TrustedAssetsInstallObserver) Observe(op gadget.ContentOperation, affectedStruct *gadget.LaidOutStructure, root, realSource, relativeTarget string) (bool, error) {
-	// TODO:UC20:
-	// steps on write action:
-	// - copy new asset to assets cache
-	// - update modeeenv
+	if affectedStruct.Role != gadget.SystemBoot {
+		// only care about system-boot
+		return true, nil
+	}
+
+	if o.bl == nil {
+		bl, err := bootloader.ForGadget(o.gadgetDir, root, &bootloader.Options{NoSlashBoot: true})
+		if err != nil {
+			return false, fmt.Errorf("cannot find bootloader: %v", err)
+		}
+		tbl, ok := bl.(bootloader.TrustedAssetsBootloader)
+		if !ok {
+			return true, nil
```

Comment:
we don't cache the bootloader on this path, maybe we should not simply not cache it?  or cache this outcome as well. Do we need anything but the bl for Name?  maybe we should cache blName either way.  In the other case we'll just have an empty o.trustedAsset and do the right thing

Reply:
Yes, at this point having the name is enough. I don't think we will need more even at later stage.

Reply:
this path is not tested btw